cli/commands/index.py

```diff
@@ -184,6 +190,87 @@ def get(
 
 
 @app.command()
-def replicate():
-    """Replicate an index from a production cluster to the rank cluster"""
-    raise NotImplementedError
+def replicate(
+    context: typer.Context,
+    source_index: str = typer.Option(
+        None,
+        help=(
+            "The name of the index to replicate. If an index is not provided, "
+            "you will be prompted to select one from the production cluster"
+        ),
+    ),
+    dest_index: str = typer.Option(
+        None,
+        help=(
+            "The name of the index to create in the rank cluster. If an index "
+            "is not provided, you will be prompted to select one from the rank "
+            "cluster. The default is to use the same name as the source index"
+        ),
+    ),
+):
+    """Reindex an index from a production cluster to the rank cluster"""
+    rank_client: Elasticsearch = context.meta["rank_client"]
+    search_templates = requests.get(
+        f"{ context.meta['catalogue_api_url']}/search-templates.json"
+    ).json()["templates"]
+
+    works = next(
+        template
+        for template in search_templates
+        if template["index"].startswith("works")
+    )
+
+    pipeline_date = re.search(
+        r"^works-indexed-(?P<date>\d{4}-\d{2}-\d{2}.?)", works["index"]
+    ).group("date")
+
+    secret_prefix = f"elasticsearch/pipeline_storage_{pipeline_date}/"
+
+    get_secret = functools.partial(aws.get_secret, context.meta["session"])
+
+    pipeline_password = get_secret(secret_prefix + "es_password")
+    pipeline_username = get_secret(secret_prefix + "es_username")
+    protocol = get_secret(secret_prefix + "protocol")
+    public_host = get_secret(secret_prefix + "public_host")
+    port = get_secret(secret_prefix + "port")
+    pipeline_host = f"{protocol}://{public_host}:{port}"
+
+    pipeline_client = elasticsearch.pipeline_client(
+        session=context.meta["session"],
+        pipeline_date=pipeline_date,
+    )
+
+    if source_index is None:
+        source_index = beaupy.select(get_valid_indices(client=pipeline_client))
+    if not rank_client.indices.exists(index=source_index):
+        raise typer.BadParameter(f"{source_index} does not exist")
+    elif source_index.startswith(".") or (source_index == "_all"):
+        raise typer.BadParameter(f"{source_index} is a system index")
+
+    if dest_index is None:
+        dest_index = typer.prompt(
+            "What do you want to call the index in the rank cluster?",
+            default=source_index,
+        )
+    raise_if_index_already_exists(context=context, index=dest_index)
+
+    task = rank_client.reindex(
```

Comment:
Is there any risk to the API cluster when we do this? I remember stuff going wrong in the past but can't remember the circumstances - might be worth having a warning "are you sure" sort of prompt.